### PR TITLE
Support for building multiarch docker images

### PR DESCRIPTION
**Pull request by klutchell**
Requires buildx for building multiarch manifests
and skopeo for copying images between the CI image
cache and the external registries.

Resolves: https://github.com/product-os/scripts/issues/155
Depends-on: https://github.com/product-os/ci-images/pull/100
Depends-on: https://github.com/product-os/balena-concourse/pull/618
Change-type: minor
Signed-off-by: Kyle Harding <kyle@balena.io>

- [x] tested with https://github.com/balenaci/vb-test/pull/139
- [x] tested with https://github.com/balenaci/librsync-go/pull/1

### CODE REVIEW

File: docker/store.sh [hunk 1]
```diff
@@ -49,11 +58,33 @@ function image_variant() {
...
+  branch_image=$(image_variant "${image}" "${branch_tag}")
+  latest_image=$(image_variant "${image}" latest)
+
+  if [ -n "${platforms}" ]
+  then
+    manifest_list=()
+    while IFS=, read -r platform
+    do
+      platform_image=$(image_variant "${image}" "${platform}")
+      import_image "${platform_image}" "${DOCKER_IMAGE_CACHE}"
```
Comment (stathismor) [inline docker/store.sh:73]: Does `.platforms` always exist, even for a single platform?
If no, then doesn't `import_image` get skipped now in the single platform case?
If yes, then can the code in `build.sh` not be written in a similar fashion, where there would always be a list of platforms and so we don't do the [if/else](https://github.com/product-os/scripts/pull/157/files#diff-efe74abc7213e95246105b4a7b029557ef16ff997e9cc13eadcfffa18e6757c0R227)?

Comment (klutchell) [reply docker/store.sh:73]: Good catch, I realized that this morning too. Platform will not always exist, I want to make sure the current behaviour doesn't change if platform isn't provided, at least for now. Maybe we will migrate all builds to buildx at a later date.

Will fix this up.

**Comment by klutchell** [comment]
What's the recommended way to test the CI locally? I could run the scripts and hardcode some stuff but I would like to see it working in a CI context to test?

**Comment by klutchell** [comment]
Running tests on this example PR: https://github.com/balenaci/vb-test/pull/139

Found that buildx wasn't installed at all so opened https://github.com/product-os/ci-images/pull/100

**Comment by klutchell** [comment]
This needs a bit of a rework as the caching and manifest creation isn't working as I expected.

I'm going to give skopeo a try so we can build once with buildx and cache in an oci dir then use skopeo copy to push it to the registry.
- https://github.com/docker/buildx/issues/430#issuecomment-887583444
- https://github.com/containers/skopeo#copying-images

**Comment by klutchell** [comment]
@balena-ci I self-certify!

**Pull request merged**